### DISCUSSION TITLE
Is there any way in which we can access solution for each indivdual time second of entire simulation time?

### DISCUSSION THREAD

**Discussion by cyasing** [pybamm-team/PyBaMM#2665] [Q&A] [answered]
I am curious if this will be possible. I am trying out a control algorithm for current control based on parameters of a cell, so I will need solution values for each second.

**Comment by Mrzhang-hub** [comment, marked as answer]
`step` method can solve your problem. Following is an example:

```
Batt_I_list = [5, 2, -4, -6]  
model = pybamm.lithium_ion.SPMe()
parameter_values = pybamm.Parameter_Values('chen2020') 
parameter_values['Current function [A]'] = '[input]'
solver = pybamm.CasadiSolver(mode = 'safe')
sim = pybamm.Simulation(model, parameter_values, solver)
for Batt_I in Batt_I_list:
  sim.step(dt=1,save=False,starting_solution=sim.solution,input={'Current function [A]': Batt_I})
  Batt_V = sim.solution['Terminal Voltage [V]'].entries
```

**Reply by cyasing** [reply]
Yes, indeed, this is the solution. Other way I found is to re-initialise the loop using `model.set_initial_conditions_from(solution_previous)` method.

**Reply by Mrzhang-hub** [reply]
Are you working on battery control problems through pybamm?  If so, we can have more deeper discussion.

**Reply by cyasing** [reply]
Not really, I'm working on optimization of battery's parameters now. I needed dynamic current control because I wanted to figure out current values based on previous second solution values of voltage in a battery pack.
